src/sqlshim.php

```diff
@@ -248,13 +248,38 @@ private static function calc_prec_scale( $prec, $scale )
   private static function typify( $row )
   {
     $i = 0;
-    foreach ( $row as $col=>&$val )
-    {
-      $val = self::guesstype($val);
-      $i++;
+    foreach ($row as &$value) {
+        //DBLIB database driver returns everything as strings, so this converts num's back to the correct data type
+        $value = self::convertDataType($value);
     }
     return $row;
   }
+  
+  private static function convertDataType($string){
```

Comment:
The only problem I have with this is the same problem I had with my own version of this function. And that's that it still doesn't know a value's actual data type in the database. You could be storing a number as text in the db and this would convert it to numeric. Correct me if I'm wrong.